File: server-py/app/models/deepseek.py
"""
模型封装
将 DeepSeek 封装为 LangChain ChatModel
DeepSeek 兼容 OpenAI 协议，使用 ChatOpenAI 并替换 base_url 即可
"""
import os
from langchain_openai import ChatOpenAI

DEEPSEEK_API_KEY = os.getenv("DEEPSEEK_API_KEY")
DEEPSEEK_BASE_URL = os.getenv("DEEPSEEK_BASE_URL", "https://api.deepseek.com/v1")
MODEL_NAME = os.getenv("MODEL_NAME", "deepseek-chat")


def create_model(**overrides) -> ChatOpenAI:
    """创建 DeepSeek 模型实例，可通过 overrides 覆盖默认参数，如 temperature=0, streaming=True"""
    params = {
        "model": MODEL_NAME,
        "api_key": DEEPSEEK_API_KEY,
        "base_url": DEEPSEEK_BASE_URL,
        "temperature": 0.7,
        "streaming": False,
    }
    params.update(overrides)
    return ChatOpenAI(**params)


# 默认导出一个标准实例（非流式）
model = create_model()

# 流式实例，用于 SSE 接口
streaming_model = create_model(streaming=True)

File: server/src/models/deepseek.js
/**
 * 第一章：模型封装
 * 将 DeepSeek 封装为 LangChain ChatModel
 * DeepSeek 兼容 OpenAI 协议，使用 ChatOpenAI 并替换 baseURL 即可
 */
import { ChatOpenAI } from '@langchain/openai';
import 'dotenv/config';

/**
 * 创建 DeepSeek 模型实例
 * @param {object} options - 可覆盖默认参数，如 { temperature: 0, streaming: true }
 */
export const createModel = (options = {}) => {
  return new ChatOpenAI({
    model: process.env.MODEL_NAME || 'deepseek-chat',
    apiKey: process.env.DEEPSEEK_API_KEY,
    configuration: {
      baseURL: process.env.DEEPSEEK_BASE_URL || 'https://api.deepseek.com/v1',
    },
    temperature: 0.7,
    streaming: false,
    ...options,
  });
};

// 默认导出一个标准实例（非流式）
export const model = createModel();

// 流式实例，用于 SSE 接口
export const streamingModel = createModel({ streaming: true });
